n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,239 @@
+"""
+The evolven3fit_API module provides the necessary tools to evolve a fit 
+using eko and to prepare it for postfit.  
+"""
+
+import pathlib
+import logging
+import numpy as np
+import yaml
+import os
+import glob
+
+from ekobox import gen_theory, gen_op, genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import run_dglap
+from eko import output
+
+from . import utils
+from validphys.loader import Loader
+
+
+logger = logging.getLogger(__name__)
+
+
+def evolve_fit(conf_folder, op_card_dict, t_card_dict, eko_path=None, dump_eko=None):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
+    """
+    log_file = pathlib.Path(conf_folder) / "evolven3fit.log"
+    if log_file.exists():
+        raise SystemError("Log file already exists: evolven3fit has already been run?")
```

Comment:
This should not be a `SystemError`, which is meant for

```
Internal error in the Python interpreter.

Please report this to the Python maintainer, along with the traceback,
the Python version, and the hardware/OS platform and version.
```

Also it is nice when scripts take care of handling errors nicely so one can differentiate between user errors such as this and code errors (see how the various other scripts we have work, from the vp main app to postfit).